python_modules/dagster/dagster/_core/execution/context/invocation.py

```diff
@@ -61,12 +60,12 @@
 
 def _property_msg(prop_name: str, method_name: str) -> str:
     return (
-        f"The {prop_name} {method_name} is not set on the context when a solid is directly invoked."
+        f"The {prop_name} {method_name} is not set on the context when an op is directly invoked."
     )
 
 
-class UnboundOpExecutionContext(OpExecutionContext):
-    """The ``context`` object available as the first argument to a solid's compute function when
+class DirectInvocationOpExecutionContext(OpExecutionContext):
```

Comment:
naming?

Reply:
i think this is fine, much clearer than previous name

Reply:
Would like to suggest `RunlessOpExecutionContext`:

- Aligns with "runless" events
- "Invocation" is not very self-explanatory and is overloaded-- it is used for instance when building a graph with the composition API inside `@job` (`PendingNodeInvocation`)

Reply:
^ I find this thinking compelling. Would also be ok with just `DirectOpExecutionContext`

Reply:
Also relevant: https://github.com/dagster-io/dagster/pull/18265#issuecomment-1839541400

Reply:
I'm not fully up to date on the runless execution work that was done, if a user wanted a context object for a runless execution is this the context they would use? If so, then I think the name `RunlessOpExecutionContext` makes sense! but if they wouldn't be using this context for a runless execution I think it could just add confusion

Reply:
The new work concerned "runless events" to my knowledge, but not "runless execution". But what we have been calling "direct invocation" is better described as "runless execution" IMO.

Reply:
ah ok my mistake. I like `RunlessOpExecution` and i don't think it will be part of the public API anyway (Bound/Unbound contexts aren't public) so even if runless events and `RunlessOpExecution` would be confusing (i don't think it is) it would have minimal exposure to users

Reply:
renamed to `RunlessOpExecution` but happy to swap names again if needed

Reply:
My initial reaction is that the overlap with "runless events" is actually confusing, since they are largely disconnect featured. I think that implies the existence of a "runless op" which isn't really true.

I would prefer nomenclature like a `PreRunOpExecutionContext`. It is the context that is created before a run exists, rather than used in a context that _permanently_ runless.

Reply:
> It is the context that is created before a run exists, rather than used in a context that permanently runless.

This class only serves direct / test execution of ops/assets in a way that I would categorize as "permanently runless". 

```
@op
def example(context):
    ...
 
example(build_op_context())
```

I am not particularly attached to runless though. Maybe
* `DirectOpExecutionContext`
* `TestOpExecutionContext`

Reply:
So contenders right now are:
* `TestOpExecutionContext`
* `DirectOpExecutionContext`
* `RunlessOpExecutionContext`


some of my thoughts:
`TestOpExecutionContext` is nice because it's explicit about when this context should be used, whereas `DirectOpExecutionContext` and `RunlessExecutionContext` require knowledge of what direct execution and runless execution are. This won't be an issue for users though since this class is not exported publicly. `DirectOpExecutionContext` lines up with our internal language for this kind of execution. I didn't find any instances of "direct invocation" in the docs though. 

I don't feel particularly strong about which name is best, but if i had to pick right now I'd probably go for `TestOpExecutionContext` then `DirectOpExecutionContext`

Reply:
>This won't be an issue for users though since this class is not exported publicly.

It isn't currently but it should be since it is returned by public functions that build test contexts.

Reply:
^^ ok makes sense

Reply:
@schrockn do you like `TestOpExecutionContext`?

Reply:
bump for @schrockn  to weigh in on naming

Reply:
Aligning on `DirectOpExecutionContext`. Reasoning:
* Some users do full integration tests (like calling `materialize`) in these cases they won't be getting a "test" context, but they are still in a testing setup. 
* `Direct` side steps this issue because it's for the case when you are "directly" providing a context object to an op/asset
* If we realize later the name needs to be different, the impact of changing the name later is low since it's a non-exported class